SU2_CFD/src/driver_structure.cpp

```diff
@@ -3414,18 +3414,35 @@ void CDriver::Interface_Preprocessing() {
         }
         /*--- Else: interpolate ---*/
         else {
-        switch (config_container[donorZone]->GetKindInterpolation()) {
+          bool conservative_interp = config_container[donorZone]->GetConservativeInterpolation();
+          
+          /*--- Conditions for conservative interpolation are not met, we cannot fallback on the consistent approach
+                because CTransfer_FlowTraction relies on the information in config to be correct. ---*/
+          if ( conservative_interp && targetZone == 0 && structural_target )
+            SU2_MPI::Error("Conservative interpolation assumes the structural model mesh is evaluated second, somehow this has not happened.",CURRENT_FUNCTION);
+        
+          switch (config_container[donorZone]->GetKindInterpolation()) {
 
           case NEAREST_NEIGHBOR:
-            interpolator_container[donorZone][targetZone] = new CNearestNeighbor(geometry_container, config_container, donorZone, targetZone);
-            if (rank == MASTER_NODE) cout << "using a nearest-neighbor approach." << endl;
-
+            if ( conservative_interp && targetZone > 0 && structural_target ) {
```

Comment:
Hi @LaSerpe , thank you for your comments.

Hence my first question, do you agree then that it makes sense to generalize the conservative interpolation for all interpolation types?

Reply:
Well, thing is that the mirror class is instantiated only in case a structural target is involved and conservative interpolation is true.
The code looks more like an override to the instantiation of the other interpolation classes than a generalisation.
I am not very familiar with fsi simulations but I do not see any reason why the user could specify an interpolation method which is then override.
Perhaps the instantiation of a CMirror class could remain an option of the switch statement?

Giulio

Reply:
Ok, then I think it won't have sense to generalise the conservative interpolation to the weighted average.
Weighted average computes the numerical fluxes between nodes and scales them by the fraction of the interface area overlapped by the neighbouring cell.
The current implementation just can't work if one of the two zones is a structure, they both have to be fluid zones.